src/device.ts

```diff
@@ -10,53 +10,58 @@
 
 'use strict';
 
-const Action = require('./action');
-const Ajv = require('ajv');
+import { Action } from "./action";
+import { Adapter } from "./adapter";
+import { Event } from "./event";
+import { Property, PropertyDescription, PropertyDict } from "./property";
+import Ajv from 'ajv';
 const ajv = new Ajv();
 
-class Device {
-  constructor(adapter, id) {
-    if (typeof id !== 'string') {
-      id = id.toString();
+export class Device {
+  public type = 'thing';
+  public '@context' = 'https://iot.mozilla.org/schemas';
+  public '@type': string[] = [];
+  public title = '';
+  public description = '';
```

Comment:
There are a lot of setters below.
Should the fields be private?

Reply:
Well, probably, but I'd imagine that would end up breaking a whole bunch of add-ons

Reply:
The variables would be still accessible for js code.
It's only a compile-time check.
I would be a good hint for the developers whether or not to access them, though.
As soon as the example adapter is converted to TypeScript the compiler would enforce it.

Reply:
Ok, that's fine then.